app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java

```diff
@@ -22,14 +25,17 @@
     
     private SendMailService sendMailService;
 
+    @Autowired
     public void setAccountDao(AccountDao accountDao) {
         this.accountDao = accountDao;
     }
     
+    @Autowired
     public void setExecutorService(ExecutorService executorService) {
         this.executorService = executorService;
     }
```

Comment:
I think it depends. For singleton classes that we define (services, controllers, DAOs), @Autowired is almost never ambiguous. For non-singleton classes (S3 clients), @Autowired is almost always ambiguous.

The ExecutorService is one of those things where right now we only have one, but I feel like it's singleton by coincidence, not by design. There's nothing stopping us from having different executors for different worker pools, so I think we should disambiguate on this.

One advantage of using @Autowired for singleton classes is if we change the implementation (fictitious example: DynamoFooDao to StormpathFooDao), @Autowired will pick that up automatically. That said, you can achieve the same effect by having a bean name "fooDao" and referencing using @Resource. (You'd just need to type a bit more.)